test/externalTests/trident.sh

```diff
@@ -0,0 +1,96 @@
+# along with solidity.  If not, see <http://www.gnu.org/licenses/>
+#
+# (c) 2021 solidity contributors.
+#------------------------------------------------------------------------------
+
+set -e
+
+source scripts/common.sh
+source test/externalTests/common.sh
+
+verify_input "$@"
+BINARY_TYPE="$1"
+BINARY_PATH="$2"
+
+function compile_fn { yarn build; }
+
+function test_fn {
+    # shellcheck disable=SC2046
+    TS_NODE_TRANSPILE_ONLY=1 npx hardhat test --no-compile $(
+        # TODO: We need to skip Migration.test.ts because it fails and makes other tests fail too.
+        # Replace this with `yarn test` once https://github.com/sushiswap/trident/issues/283 is fixed.
+        find test/ -name "*.test.ts" ! -path "test/Migration.test.ts"
+    )
+}
+
+function trident_test
+{
+    local repo="https://github.com/sushiswap/trident"
+    local branch=master
+    local config_file="hardhat.config.ts"
+    local config_var=config
+
+    local compile_only_presets=()
+    local settings_presets=(
+        "${compile_only_presets[@]}"
+        #ir-no-optimize            # Compilation fails with: "YulException: Variable var_amount_165 is 9 slot(s) too deep inside the stack."
+        #ir-optimize-evm-only      # Compilation fails with: "YulException: Variable var_amount_165 is 9 slot(s) too deep inside the stack."
+        #ir-optimize-evm+yul       # Compilation fails with: "YulException: Cannot swap Variable var_nearestTick with Variable _4: too deep in the stack by 4 slots"
+        legacy-no-optimize
+        legacy-optimize-evm-only
+        legacy-optimize-evm+yul
+    )
+
+    local selected_optimizer_presets
+    selected_optimizer_presets=$(circleci_select_steps_multiarg "${settings_presets[@]}")
+    print_optimizer_presets_or_exit "$selected_optimizer_presets"
+
+    setup_solc "$DIR" "$BINARY_TYPE" "$BINARY_PATH"
+    download_project "$repo" "$branch" "$DIR"
+
+    # TODO: Currently tests work only with the exact versions from yarn.lock.
+    # Re-enable this when https://github.com/sushiswap/trident/issues/284 is fixed.
+    #neutralize_package_lock
+
+    neutralize_package_json_hooks
+    force_hardhat_compiler_binary "$config_file" "$BINARY_TYPE" "$BINARY_PATH"
+    force_hardhat_compiler_settings "$config_file" "$(first_word "$selected_optimizer_presets")" "$config_var"
+    yarn install
+
+    replace_version_pragmas
+    force_solc_modules "${DIR}/solc"
+
+    # BentoBoxV1Flat.sol requires a few small tweaks to compile on 0.8.x.
+    # TODO: Remove once https://github.com/sushiswap/trident/pull/282 gets merged.
+    sed -i 's|uint128(-1)|type(uint128).max|g' contracts/flat/BentoBoxV1Flat.sol
+    sed -i 's|uint64(-1)|type(uint64).max|g' contracts/flat/BentoBoxV1Flat.sol
+    sed -i 's|uint32(-1)|type(uint32).max|g' contracts/flat/BentoBoxV1Flat.sol
+    sed -i 's|IERC20(0)|IERC20(address(0))|g' contracts/flat/BentoBoxV1Flat.sol
+    sed -i 's|IStrategy(0)|IStrategy(address(0))|g' contracts/flat/BentoBoxV1Flat.sol
```

Comment:
Oh wow!

Reply:
Mudit said bentobox is a legacy deployed contract, and they probably won't change it in the repo.

I just checked that there are no contract dependencies for Bento. So the project will still compile without this file, however, it's needed for tests. Fine with the sed.

Reply:
Yeah, unfortunately we really do want to run tests, not just compile. Especially now that #12441 requires a full test run to gather gas usage for the benchmark.